File: apps/store-desktop/src/main/paths.ts
import os from 'node:os';
import path from 'node:path';

export const ULANZI_APP = 'Ulanzi Studio';

export function pluginsDir(): string {
  if (process.env.STORE_PLUGINS_DIR) return process.env.STORE_PLUGINS_DIR;
  if (process.platform === 'darwin') {
    return path.join(os.homedir(), 'Library', 'Application Support', 'Ulanzi', 'UlanziDeck', 'Plugins');
  }
  if (process.platform === 'win32') {
    const appData = process.env.APPDATA || path.join(os.homedir(), 'AppData', 'Roaming');
    return path.join(appData, 'Ulanzi', 'UlanziDeck', 'Plugins');
  }
  return path.join(os.homedir(), '.ulanzideck', 'Plugins');
}

File: helper/paths.go
package main

import (
	"os"
	"path/filepath"
)

// pluginsDir é onde o UlanziDeck lê os plugins instalados (macOS).
// Pode ser sobrescrito com HELPER_PLUGINS_DIR (útil para testes/CI e configs futuras).
func pluginsDir() string {
	if v := os.Getenv("HELPER_PLUGINS_DIR"); v != "" {
		return v
	}
	home, _ := os.UserHomeDir()
	return filepath.Join(home, "Library", "Application Support", "Ulanzi", "UlanziDeck", "Plugins")
}

// stateDir guarda dados do helper (tmp de download, registro de origem dos plugins).
func stateDir() string {
	home, _ := os.UserHomeDir()
	return filepath.Join(home, ".ulanzideck-store")
}

func ensureDir(p string) error {
	return os.MkdirAll(p, 0o755)
}
